iam_redpanda_agent.tf

```diff
@@ -695,6 +696,7 @@ data "aws_iam_policy_document" "agent_permissions_boundary_scoped_iam" {
       "arn:aws:iam::${local.aws_account_id}:role/redpanda-*-redpanda-connect",
       "arn:aws:iam::${local.aws_account_id}:role/redpanda-*-redpanda-connect-pipeline",
       "arn:aws:iam::${local.aws_account_id}:role/redpanda-*-operator-role",
+      "arn:aws:iam::${local.aws_account_id}:role/redpanda-*-redpanda-connect-mcp_server"
```

Comment:
We could consider changing these all to `redpanda-*` to make future changes more seamless.